src/doom/am_map.c

```diff
@@ -1766,10 +1769,15 @@ static void AM_drawWalls (void)
                         if (!lines[i].backsector)
                         {
                             // [JN] Highlight secret sectors
-                            if (automap_secrets && lines[i].frontsector->special == 9)
+                            if (automap_secrets > 1 && lines[i].frontsector->special == 9)
                             {
                                 AM_drawMline(&l, secretwallcolors);
                             }
+                            // [plums] show revealed secrets
+                            else if (automap_secrets && lines[i].frontsector->oldspecial == 9)
+                            {
+                                AM_drawMline(&l, foundsecretwallcolors);
```

Comment:
Uh-oh, there might be a problem with merging. Have a look please, I recently [updated](https://github.com/JNechaevsky/international-doom/commit/385c0700896f7a34143ceae00661827822a06336) walls drawing function to use `array_push` in all three games: https://github.com/JNechaevsky/international-doom/blob/main/src/doom/am_map.c#L1779-L1782

Reply:
Yeah that will definitely be a problem. I had hoped github would have flagged that as a merge conflict, but it didn't. 

I'll address that soon, thanks for checking it.